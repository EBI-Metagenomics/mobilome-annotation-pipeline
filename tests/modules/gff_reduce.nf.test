nextflow_process {

    name "Test Process GFF_REDUCE"
    script "modules/gff_reduce.nf"
    process "GFF_REDUCE"

    test("Should process the mobilome prokka gff file") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/test_data/mobilome_results/mobilome_prokka.gff")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
