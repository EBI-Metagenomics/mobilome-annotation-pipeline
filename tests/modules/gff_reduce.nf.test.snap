{
    "Should process the mobilome prokka gff file": {
        "content": [
            {
                "0": [
                    "mobilome_clean.gff:md5,921a580f47ae19d8d322c0c651c2ce35"
                ],
                "1": [
                    "mobilome_extra.gff:md5,e47629cbc74dc1ad1f75a737b41a86ff"
                ],
                "2": [
                    "mobilome_nogenes.gff:md5,ccb81bb84e850a93dd5e819a0880781f"
                ],
                "mobilome_clean": [
                    "mobilome_clean.gff:md5,921a580f47ae19d8d322c0c651c2ce35"
                ],
                "mobilome_extra": [
                    "mobilome_extra.gff:md5,e47629cbc74dc1ad1f75a737b41a86ff"
                ],
                "mobilome_nogenes": [
                    "mobilome_nogenes.gff:md5,ccb81bb84e850a93dd5e819a0880781f"
                ]
            }
        ],
        "timestamp": "2024-05-29T12:52:25.795589494"
    }
}